api/src/main/java/ai/djl/ndarray/NDArray.java

```diff
@@ -4543,4 +4543,25 @@ default NDArray norm(int[] axes, boolean keepDims) {
      * @return the norm of this {@code NDArray}
      */
     NDArray norm(int ord, int[] axes, boolean keepDims);
+
+    /**
+     * Returns a one-hot {@code NDArray}.
+     *
+     * @param depth Depth of the one hot dimension.
+     * @param onValue The value assigned to the locations represented by indices.
+     * @param offValue The value assigned to the locations not represented by indices.
+     * @param dataType dataType of the output.
+     * @return one-hot encoding of this {@code NDArray}
+     */
+    NDArray oneHot(int depth, float onValue, float offValue, DataType dataType);
+
+    /**
+     * Returns a one-hot {@code NDArray}.
+     *
+     * @param depth Depth of the one hot dimension.
```

Comment:
Can you reverse the order on these? We tend to have the helpers at the top and the base implementation at the bottom

Reply:
Alright. Done